habitat/tasks/nav/nav_task.py

```diff
@@ -204,6 +207,84 @@ def get_observation(self, observations, episode):
         return direction_vector_agent
 
 
+class StaticPointGoalSensor(habitat.Sensor):
+    """
+    Sensor for PointGoal observations which are used in the StaticPointNav task.
+    For the agent in simulator the forward direction is along negative-z.
+    In polar coordinate format the angle returned is azimuth to the goal.
+    Args:
+        sim: reference to the simulator for calculating task observations.
+        config: config for the PointGoal sensor. Can contain field for
+            GOAL_FORMAT which can be used to specify the format in which
+            the pointgoal is specified. Current options for goal format are
+            cartesian and polar.
+    Attributes:
+        _goal_format: format for specifying the goal which can be done
+            in cartesian or polar coordinates.
+    """
+
+    def __init__(self, sim, config):
+        self._sim = sim
+        self._goal_format = getattr(config, "GOAL_FORMAT", "CARTESIAN")
+        assert self._goal_format in ["CARTESIAN", "POLAR"]
+
+        super().__init__(sim, config)
+        self.initial_vector = None
+        self.episode_unique_id = None
+
+    def _get_uuid(self, *args: Any, **kwargs: Any):
+        return "static_pointgoal"
+
+    def _get_sensor_type(self, *args: Any, **kwargs: Any):
+        return SensorTypes.STATIC_GOAL_VECTOR
+
+    def _get_observation_space(self, *args: Any, **kwargs: Any):
+        if self._goal_format == "CARTESIAN":
+            sensor_shape = (3,)
+        else:
+            sensor_shape = (2,)
+        return spaces.Box(
+            low=np.finfo(np.float32).min,
+            high=np.finfo(np.float32).max,
+            shape=sensor_shape,
+            dtype=np.float32,
+        )
+
+    def get_observation(self, observations, episode):
+        episode_unique_id = (episode.episode_id, episode.scene_id)
```

Comment:
replace `episode_unique_id` with `episode_id` and `self.episode_unique_id` with `self.current_episode_id`